spec/spec_helper.rb

```diff
@@ -14,24 +14,41 @@
   end
 end
 
+TEST_CREDENTIALS_DEFAULTS = {
+  :vmware_cloud_defaults => {:host => "vmwarecloudhost", :userid => "VMWARE_CLOUD_USERID", :password => "VMWARE_CLOUD_PASSWORD"},
+  :vmware_infra_defaults => {:hostname => "HOSTNAME"},
+  :vmware_tanzu_defaults => {:hostname => "vmware-tanzu-hostname", :userid => "VMWARE_TANZU_USERID", :password => "VMWARE_TANZU_PASSWORD"}
+}.freeze
+
+def test_credentials(*args)
+  Rails.application.credentials.dig(*args) || test_credentials_defaults(*args)
+end
+
+def test_credentials_defaults(*args)
+  args[0] = "#{args[0]}_defaults".to_sym
+  TEST_CREDENTIALS_DEFAULTS.dig(*args)
+end
+
 VCR.configure do |config|
+  # config.default_cassette_options = { :record => :all }
```

Comment:
placeholder to verify cassette recording uses the correct values from local rails credentials and falls back to the defaults.